examples/Example11/electrons.cu

```diff
@@ -0,0 +1,255 @@
+    // which might be longer than the geometrical step length due to MSC. As
+    // long as we call PerformContinuous in the same kernel we don't need to
+    // care, but we need to make this available when splitting the operations.
+    // double physicalStepLength = elTrack.GetPStepLength();
+    int winnerProcessIndex = theTrack->GetWinnerProcessIndex();
+    // Leave the range and MFP inside the G4HepEmTrack. If we split kernels, we
+    // also need to carry them over!
+
+    // Check if there's a volume boundary in between.
+    double geometryStepLength = fieldPropagatorBz.ComputeStepAndPropagatedState</*Relocate=*/false>(
+        currentTrack.energy, Mass, Charge, geometricalStepLengthFromPhysics, currentTrack.pos, currentTrack.dir,
+        currentTrack.currentState, currentTrack.nextState);
+
+    if (currentTrack.nextState.IsOnBoundary()) {
+      theTrack->SetGStepLength(geometryStepLength);
+      theTrack->SetOnBoundary(true);
+    }
+
+    // Apply continuous effects.
+    bool stopped = G4HepEmElectronManager::PerformContinuous(&g4HepEmData, &g4HepEmPars, &elTrack);
+    // Collect the changes.
+    currentTrack.energy = theTrack->GetEKin();
+    atomicAdd(&scoring->energyDeposit, theTrack->GetEnergyDeposit());
+
+    // Save the `number-of-interaction-left` in our track.
+    for (int ip = 0; ip < 3; ++ip) {
+      double numIALeft           = theTrack->GetNumIALeft(ip);
+      currentTrack.numIALeft[ip] = numIALeft;
+    }
+
+    if (stopped) {
+      if (!IsElectron) {
+        // Annihilate the stopped positron into two gammas heading to opposite
+        // directions (isotropic).
+        Track &gamma1 = secondaries.gammas.NextTrack();
+        Track &gamma2 = secondaries.gammas.NextTrack();
+        atomicAdd(&scoring->secondaries, 2);
+
+        const double cost = 2 * currentTrack.Uniform() - 1;
+        const double sint = sqrt(1 - cost * cost);
+        const double phi  = k2Pi * currentTrack.Uniform();
+        double sinPhi, cosPhi;
+        sincos(phi, &sinPhi, &cosPhi);
+
+        gamma1.InitAsSecondary(/*parent=*/currentTrack);
+        gamma1.rngState = currentTrack.rngState.Branch();
+        gamma1.energy = copcore::units::kElectronMassC2;
+        gamma1.dir.Set(sint * cosPhi, sint * sinPhi, cost);
+
+        gamma2.InitAsSecondary(/*parent=*/currentTrack);
+        // Reuse the RNG state of the dying track.
+        gamma2.rngState = currentTrack.rngState;
+        gamma2.energy = copcore::units::kElectronMassC2;
+        gamma2.dir    = -gamma1.dir;
+      }
+      // Particles are killed by not enqueuing them into the new activeQueue.
+      continue;
+    }
+
+    if (currentTrack.nextState.IsOnBoundary()) {
+      // For now, just count that we hit something.
+      atomicAdd(&scoring->hits, 1);
+
+      BVHNavigator::RelocateToNextVolume(currentTrack.pos, currentTrack.dir, currentTrack.nextState);
+
+      // Kill the particle if it left the world.
+      if (currentTrack.nextState.Top() != nullptr) {
```

Comment:
The problem is that we pass in `nextState` and right before leaving (i.e. within world volume and on the way out), the next state may be already outside, I guess. For `nextState.Top()` to be `nullptr` at this point, it could already have been `nullptr` before the call above. In any case, I think the change to `RelocateToNextVolume` is needed, what may need checking again is the place where the call to relocate is made, and if `nextState` is the right thing to pass in as argument (I used your reference commit to decide this).

Reply:
`*Navagator::ComputeStepAndNextVolume` pops one volume from `nextState` because that's where we know that we're leaving the current volume to its parent. This should work if the track was previously in the world volume. But now that I think about it, I wonder what happens if a volume is right at the world's boundary. Probably crashes, but we should properly fix this in a separate PR

Reply:
Still in the wrong place AFAICT

Reply:
As I said, I used your commit as reference to decide where to put the call. I didn't think it would matter that much because in your version of the code the relocation happens after the whole kernel is done anyways. Maybe I should just put the call to relocate after `currentTrack.SwapStates();`? Just telling me it's in the wrong place doesn't help much. It's probably easier for you to fix this (since you wrote the rest of this example) by moving the calls where they belong after this is merged.

Reply:
Please have a look in the current `example9`:
https://github.com/apt-sim/AdePT/blob/f413adfaacc355913229713b6ce6e7254faed856/examples/Example9/electrons.cu#L128-L135

You're doing it differently in this example, and that's why it's crashing.

Reply:
I moved the call, now it works.